src/components/login-dropdown.tsx

```diff
@@ -25,6 +26,71 @@ const ChaiseLoginDropdown = ({
   menu, openProfileCb, parentDropdown
 }: ChaiseLoginDropdownProps): JSX.Element => {
   const dropdownWrapper = useRef<any>(null); // TODO: type the useRef wrapped element
+  const subMenuRef = useRef<any>(null);
+
+  // state variable to define left, top, dropEnd or dropStart positions...
+  const [fromTop, setFromTop] = useState<number>();
+  const [fromLeft, setFromLeft] = useState<number>();
+  const [dropEnd, setDropEnd] = useState<boolean>(true);
+
+  useLayoutEffect(() => {
+    
+    // when resize event is called, it will call debounce function with 500ms timeout
+    const debouncedFunc = debounce(setHeight, 500);
+    
+    // First time it will be called diretly
+    setHeight();
+
+    window.addEventListener('resize', debouncedFunc);
+  })
+
+  const setHeight = () => {
+    const winHeight = windowRef.innerHeight;
+    const padding = 15;
+
+    if (subMenuRef && subMenuRef.current) {
+      // This logic will run when there is no resize event, but called on click of menu item
+      const y = subMenuRef.current.getBoundingClientRect().y;
+      const available = winHeight - y;
+
+      subMenuRef.current.style.maxHeight = available - padding + 'px';
+    } else {
```

Comment:
Why do you need this else case here?

Reply:
When multiple submenus are open on the window, and resize event happens, it should calculate height of all submenus & update.

Reply:
If the “else”  is handling all the submenus, then what’s the point of the “if” case? and if it’s general and for all the submenus, it should go to the navbar.tsx file and the selector should be specific to only navbar submenus and not all the dropdown menus